src/main/java/org/deepsymmetry/beatlink/DeviceFinder.java

```diff
@@ -180,268 +134,232 @@ public boolean isAddressIgnored(InetAddress address) {
-                }
+            final boolean foundNewDevice = isDeviceNew(announcement);
+            updateDevices(announcement);
+            if (foundNewDevice) {
+                deliverFoundAnnouncement(announcement);
+            }
+            if (VirtualCdj.getInstance().isRunning() &&
+                    announcement.getDeviceNumber() == VirtualCdj.getInstance().getDeviceNumber()) {
+                // Someone is using the same device number as we are! Try to defend it.
+                VirtualCdj.getInstance().defendDeviceNumber(announcement.getAddress());
             }
-        });
-    }
-
-    /**
-     * Stop listening for device announcements. Also discard any announcements which had been received, and
-     * notify any registered listeners that those devices have been lost.
-     */
-    @SuppressWarnings("WeakerAccess")
-    public synchronized void stop() {
-        if (isRunning()) {
-            socket.get().close();
-            socket.set(null);
-            flush();
-            deliverLifecycleAnnouncement(logger, false);
         }
-    }
 
-    /**
-     * Get the set of DJ Link devices which currently can be seen on the network. These can be passed to
-     * {@link VirtualCdj#getLatestStatusFor(DeviceUpdate)} to find the current detailed status for that device,
-     * as long as the Virtual CDJ is active.
-     *
-     * @return the devices which have been heard from recently enough to be considered present on the network
-     *
-     * @throws IllegalStateException if the {@code DeviceFinder} is not active
-     */
-    public Set<DeviceAnnouncement> getCurrentDevices() {
-        if (!isRunning()) {
-            throw new IllegalStateException("DeviceFinder is not active");
-        }
-        expireDevices();  // Get rid of anything past its sell-by date.
-        // Make a copy so callers get an immutable snapshot of the current state.
-        return Collections.unmodifiableSet(new HashSet<DeviceAnnouncement>(devices.values()));
+        expireDevices();
     }
 
-    /**
-     * Find and return the device announcement that was most recently received from a device identifying itself
-     * with the specified device number, if any.
-     *
-     * @param deviceNumber the device number of interest
-     *
-     * @return the matching announcement or null if no such device has been heard from
-     *
-     * @throws IllegalStateException if the {@code DeviceFinder} is not active
-     */
-    public DeviceAnnouncement getLatestAnnouncementFrom(int deviceNumber) {
-        ensureRunning();
-        for (DeviceAnnouncement announcement : getCurrentDevices()) {
-            if (announcement.getDeviceNumber() == deviceNumber) {
-                return announcement;
+/**
+ * Discard any knowledge we have about current devices. Called when shutting down, and also by the
+ * {@link VirtualCdj} when it believes the network has changed in a way that makes them no longer
+ * reachable.
+ */
+synchronized void flush() {
```

Comment:
Oh, this is what confused me in the other class. `flush()` to me means to ensure that a file/socket/etc. has written its buffers to its destination. This will go back to being `stop()` when we restore the start/stop lifecycle of this class.